ui/component/searchOptions/view.jsx

```diff
@@ -16,6 +16,11 @@ const SearchOptions = (props: Props) => {
   const { options, setSearchOption, expanded, toggleSearchExpanded } = props;
   const resultCount = options[SEARCH_OPTIONS.RESULT_COUNT];
 
+  const isFilteringByChannel = useMemo(() => {
+    const claimType = String(options[SEARCH_OPTIONS.CLAIM_TYPE] || '');
+    return claimType.includes(SEARCH_OPTIONS.INCLUDE_CHANNELS);
+  }, [options]);
```

Comment:
To make sure this works properly, you need to stringify the object that you pass into a react hook

I usually do something like
```
const strinigifiedOptions = JSON.stringify(options)

React.useEffect(() => {
  const jsonOptions = JSON.parse(stringifiedOptions);
  ...
  }, [stringifiedOptions])
```

Reply:
I believe this is hidding a bug. We shouldn't need to `stringify` the object unless it's being mutated. Will update though but, I think this is a bug.

Reply:
It's just how react hooks work. Objects are special. Even if you don't mutate the object it still needs to be stringified.

Reply:
According to the documentation:

https://reactjs.org/docs/hooks-reference.html#usememo

> Pass a “create” function and an array of dependencies. useMemo will only recompute the memoized value when one of the dependencies has changed. This optimization helps to avoid expensive calculations on every render.

Since props are immutable, it means it will only check if it's filtering by channel when `props.options` change. This is why I was asking if `options` is immutable, because if it is, there is no need for `stringify`. If it isn't, we may need to dig deeper since it may cause issues.